### PR TITLE
New Exercise: diamond

### PR DESCRIPTION
**Pull request by jvarness**
[Rihanna - Diamond](https://www.youtube.com/watch?v=lWA2pjMjpBs)

### CODE REVIEW

**Comment by jvarness** [comment]
Does the config look ok to you guys? Correct difficulty? Correct topics? Needs to be `unlocked_by` something or not?

**Comment by Stargator** [comment]
You should run presubmit

File: config.json [hunk 1]
```diff
@@ -193,6 +193,18 @@
         "equality",
         "strings"
       ]
+    },
+    {
+      "slug": "diamond",
+      "uuid": "5bbd251d-b2eb-44a1-a9a6-8543afb7511c",
+      "core": false,
```
Comment (devkabiir) [inline config.json]: If it's not `core` then it should have an `unlocked_by` right? 🤔

File: exercises/diamond/test/diamond_test.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import 'package:diamond/diamond.dart';
+
+void main() {
+  final diamond = new Diamond();
+
+  group('Diamond', () {
+    test("Degenerate case with a single 'A' row", () {
+      final List<String> result = diamond.rows("A");
+      expect(result, equals(["A"]));
+    }, skip: false);
```
Comment (devkabiir) [inline exercises/diamond/test/diamond_test.dart]: All of these `skip` should be `true`

File: config.json (hunk 1 above)
Comment (Stargator) [reply config.json]: Yes

File: exercises/diamond/test/diamond_test.dart (hunk 2 above)
Comment (Stargator) [reply exercises/diamond/test/diamond_test.dart]: Except the first test case

File: config.json (hunk 1 above)
Comment (jvarness) [reply config.json]: Which exercise should it be unlocked by? Or are you saying it needs to just be marked as core?

File: exercises/diamond/test/diamond_test.dart (hunk 2 above)
Comment (jvarness) [reply exercises/diamond/test/diamond_test.dart]: Resolved in 12e3472

File: config.json (hunk 1 above)
Comment (Stargator) [reply config.json]: @jvarness I'm going to suggest making it `core` after Anagram. That means setting core as false and unlocked as null.

@devkabiir Thoughts?

File: exercises/diamond/test/diamond_test.dart [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import 'package:test/test.dart';
+import 'package:diamond/diamond.dart';
+
+void main() {
+  final diamond = new Diamond();
+
+  group('Diamond', () {
+    test("Degenerate case with a single 'A' row", () {
+      final List<String> result = diamond.rows("A");
```
Comment (Stargator) [inline exercises/diamond/test/diamond_test.dart]: @jvarness Where possible can you use single quotes (`'`)?

**Comment by Stargator** [comment]
@jvarness I appreciate your work, it looks great. Just one request, for future pull requests, keep your branches in your fork. Since I have your forked dart repo in my local remotes, it can be confusing which one is the branch to be reviewed.

Plus, I think it just adds more work for you to sync between your forked repo and this repo.

**Comment by jvarness** [comment]
@Stargator I can do that, but doesn't managing the forks make things a little difficult to manage? Part of the reason why I wanted to become a maintainer of this repo was so that I didn't have to manage forks. I wanted to be able to just keep track of my branches.

I'm fine with using a fork, but for me it results in a lot less work keeping my fork up-to-date.

**Comment by Stargator** [comment]
@jvarness That works too. I think my intent was to have the branch for the work only exist in one repo.

**Comment by Stargator** [comment]
@jvarness I'm leaving it up to you to decide whether the diamond branch is ready to be deleted.

**Comment by jvarness** [comment]
It is :) deleted it!

**Pull request merged**